processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -40,8 +46,10 @@
   private final String lower;
   private final boolean lowerStrict;
   private final boolean upperStrict;
-  private final boolean alphaNumeric;
+  private final Boolean alphaNumeric;
```

Comment:
alphaNumeric doesn't need to be a class field, it only needs to be understood in the constructor and translated to an ordering as appropriate (allowing us to migrate parameters this way is why http://druid.io/docs/latest/operations/rolling-updates.html recommends updating historical/MMs before brokers).